subsys/usb/class/hid/Kconfig

```diff
@@ -44,12 +44,18 @@ config USB_HID_POLL_INTERVAL_MS
 	  Polling interval in ms selected by the USB HID Device.
 
 config USB_HID_REPORTS
-	int "HID reports in the instance"
+	int "Number of HID idle reports in the instance"
 	default 1
-	range 1 256
+	range 1 8
```

Comment:
Is there a reason why the range is maxed at 8?
On a single device you can have multiple input reports as well as output and feature reports. Since there is no mapping report's id is used to index the array element. In case of multiple reports 8 slots can be just not enough. Integrator may be willing to sacrifice some memory to fit more reports in the map.

If my understanding is correct the idle report feature regards only input reports. Is this right?

Reply:
> Is there a reason why the range is maxed at 8?
> On a single device you can have multiple input reports as well as output and feature reports. Since there is no mapping report's id is used to index the array element. In case of multiple reports 8 slots can be just not enough. Integrator may be willing to sacrifice some memory to fit more reports in the map.

One can have as many ReportIDs as possible, for example 42 or none and report everything at once. This configuration value is about "Number of HID idle reports", what is really rarely used, and there is no way to have offsets or gaps in callback ReportIDs for that with current HID implementation. So this feature is limited anyway, and large values can cause wrong expectations.

Bluetooth has removed/deprecated Set/Get Idle from HID Profile spec, what I think is very reasonable.

> 
> If my understanding is correct the idle report feature regards only input reports. Is this right?

Yes, only IN endpoints and and therefore only input reports.

Reply:
quick question
If I will define that first 8th reports are output and 9th is input and idle report, will the code handle it? this report will be referenced with id=8?

Reply:
it's possible that it is somewhere in the code but I don't see mapping between report_id and idx of array; when arrays are used there is always `xyz[report_id]` which would indicate that reports placed later (beyond the max range) cannot use this feature.

Reply:
> quick question
> If I will define that first 8th reports are output and 9th is input and idle report, will the code handle it? this report will be referenced with id=8?

No, and not before this patch series, offsets and gaps can not be handled. In your example CONFIG_USB_HID_REPORTS would be 1, because you have only **one Input Item**. The ID in the idle callback would be 1 and is probably not what application expects, but Get/SetIdle would  have ReportID = 8 as parameter. We can not handle that with current implementation and API, that needs much more rework. The effort to implement it properly is inversely proportional to the benefit.

Reply:
>  offsets and gaps can not be handled
What I meant is that one can define a report map with all 8 first report ids given to output reports and input report being given id 9. All reports from 1-9 will be used (so no gap in ids), but not all for input reports.
When talking about gap, do you mean that lowest report ids must be dedicated to the input reports that use the feature. So arrays get the consecutive set of reports using the feature (from 1 to n). Is this right?

That last thing is a limitation but maybe not so important.

Going back to my question. Why limit it to 8? Won't it be better for integrator to decide? (Or maybe there is some bitmask somewhere on u8 that I have missed).

Reply:
> When talking about gap, do you mean that lowest report ids must be dedicated to the input reports that use the feature. So arrays get the consecutive set of reports using the feature (from 1 to n). Is this right?

Yes.

> Going back to my question. Why limit it to 8? Won't it be better for integrator to decide? (Or maybe there is some bitmask somewhere on u8 that I have missed).

There is no special reason for 8, I thought it may help to understand the problem discussed above :smile:. Which maximal value do you prefer?

Reply:
Maybe leave it as it was?

Reply:
okay, [done](https://github.com/zephyrproject-rtos/zephyr/compare/6e79800df0e0efdd0808935a1e7b1fe173c703bb..ea2e5eafa0926120192a083c0af94bfb440c941e)